contracts/0.4.24/Lido.sol

```diff
@@ -999,10 +790,65 @@ contract Lido is ILido, StETH, AragonApp {
     }
 
     /**
-    * @dev Size-efficient analog of the `auth(_role)` modifier
-    * @param _role Permission name
-    */
+     * @dev Size-efficient analog of the `auth(_role)` modifier
+     * @param _role Permission name
+     */
     function _auth(bytes32 _role) internal view auth(_role) {
         // no-op
     }
+
+    function _transferToStakingRouter(uint256 _maxDepositsCount) internal {
+        address stakingRouter = getStakingRouter();
+        require(stakingRouter != address(0), "STAKING_ROUTER_ADDRESS_ZERO");
+
+        uint256 buffered = _getBufferedEther();
+        if (buffered >= DEPOSIT_SIZE) {
+            uint256 unaccounted = _getUnaccountedEther();
+            uint256 numDeposits = buffered.div(DEPOSIT_SIZE);
+            numDeposits = numDeposits < _maxDepositsCount ? numDeposits : _maxDepositsCount;
+
+            uint256 amount = numDeposits * DEPOSIT_SIZE;
+
+            address(stakingRouter).transfer(amount);
+
+            BUFFERED_ETHER_POSITION.setStorageUint256(BUFFERED_ETHER_POSITION.getStorageUint256().sub(amount));
+
+            STAKING_ROUTER_BUFFERED_ETHER_POSITION.setStorageUint256(
+                STAKING_ROUTER_BUFFERED_ETHER_POSITION.getStorageUint256().add(amount)
+            );
+
+            emit Unbuffered(amount);
+
+            assert(_getUnaccountedEther() == unaccounted);
+        }
+    }
+
+    /**
+     * @dev Invokes a deposit call to the Staking Router contract and updates buffered counters
+     * @param _maxDepositsCount max deposits count
+     * @param _stakingModuleId id of the staking module to be deposited
+     * @param _depositCalldata module calldata
+     */
+    function deposit(uint256 _maxDepositsCount, uint24 _stakingModuleId, bytes _depositCalldata) external whenNotStopped {
+        require(msg.sender == getDepositSecurityModule(), "APP_AUTH_DSM_FAILED");
+
+        //make buffer transfer from LIDO to StakingRouter
+        _transferToStakingRouter(_maxDepositsCount);
```

Comment:
transfer `getBufferedEther` and not only `_maxDepositsCount `

Reply:
fixed